### PR TITLE
Bump golang 1.18 and k8s 1.24

### PR DESCRIPTION
**Pull request by SchSeba**
Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

**Comment by adrianchiris** [comment]
need also to update documentation to say we support go version 1.16+ to match what we test building against

File: .github/workflows/buildtest.yml [hunk 1]
```diff
@@ -9,15 +9,15 @@ jobs:
   build:
     strategy:
       matrix:
-        go-version: [1.13.x, 1.14.x, 1.15.x, 1.16.x]
+        go-version: [1.16.x, 1.17.x, 1.18.x]
```
Comment (adrianchiris) [inline .github/workflows/buildtest.yml:12]: shall we test build against 1.17.x and 1.18.x like other projects ?

Comment (SchSeba) [reply .github/workflows/buildtest.yml:12]: yep I miss that

**Comment by SchSeba** [comment]
done

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**